lib/runner.js

```diff
@@ -814,6 +814,14 @@ Runner.prototype.runTests = function (suite, fn) {
             self.fail(test, err);
           }
           self.emit(constants.EVENT_TEST_END, test);
+          return self.hookUp(HOOK_TYPE_AFTER_EACH, next);
+        } else if (test.currentRepeat() < test.repeats()) {
+          var repeatedTest = test.clone();
+          repeatedTest.currentRepeat(test.currentRepeat() + 1);
+          tests.unshift(repeatedTest);
+
+          self.emit(constants.EVENT_TEST_RETRY, test, null);
```

Comment:
I agree (especially since `EVENT_TEST_RETRY` is supposed to have a mandatory `Error` argument) - but I wonder whether there should be a special event. You will get the `EVENT_TEST_BEGIN` /  `EVENT_TEST_END` / `EVENT_TEST_PASS` chain anyway. Should there be another event after these when a test is to be repeated? Should it come in place of the `EVENT_TEST_PASS`?

Reply:
Hmm yeah that's a good point. I think it'd be reasonable to leave out the event for now. We can always add it in if folks ask. Taking one out is much harder.

Will also want to hear from @mochajs/maintenance-crew on this (and the PR in general, per the review note).

Reply:
Oh, and per https://github.com/mochajs/mocha/pull/5011/#issuecomment-1942519084: if there's no event then I think it'd be especially valuable to make sure test contexts have the current repeat count.